lib/src/converter.dart

```diff
@@ -36,14 +37,18 @@ final _trailingNewLinesRegExp = RegExp(r'\n*$');
 /// The [rules] parameter can be used to customize element processing.
 ///
 String convert(
-  String html, {
+  dynamic input, {
   String? rootTag,
   String? imageBaseUrl,
   Map<String, String>? styleOptions,
   List<String>? ignore,
   List<Rule>? rules,
 }) {
-  if (html.isEmpty) {
+  // the input can be html string or dom.Node
+  if (input is! String && input is! dom.Node) {
+    return '';
+  }
+  if (input is String && input.isEmpty) {
```

Comment:
merge into one condition? `if (input is String && input.isEmpty || input is! dom.Noe)`